### DISCUSSION TITLE
ItemStack#getTag() returns null even if the tag is there

### DISCUSSION THREAD

**Discussion by Tofaa2** [Minestom/Minestom#1262] [Help] [answered]
Steps used to replicate the following null pointer:

java.lang.NullPointerException: Cannot invoke "java.lang.Double.intValue()" because the return value of "net.minestom.server.item.ItemStack.getTag(net.minestom.server.tag.Tag)" is null

Ignore the .intValue() boxing that was just a test.
```java
    public @NotNull Result run(@NotNull PlayerChangeHeldSlotEvent event) {
        byte slot = event.getSlot();
        ItemStack stack = event.getPlayer().getInventory().getItemStack(slot); 
        if (stack.isAir()) {
            return Result.SUCCESS;
        }

        Highworld.getInstance().getPlayerDataManager().getPlayerData(event.getPlayer()).getAttributeProcessor().refreshStats(stack);
        return Result.SUCCESS;
    }

```

The method AttributeProcessor#refreshStats(ItemStack item) :
```java

public void refreshStats(ItemStack held) {
    if (held == null) held = player.getItemInMainHand();
    if (!held.isAir()) {

       if (held.hasTag(Tag.Double("strength") {
           player.sendMessage("Mogus"); // This does get sent to the player
           strength += player.getItemInMainHand().getTag(Tag.Double("strength")).intValue(); // This is what the error is referring to 
        }
    }
}

**Comment by ZakShearman** [comment, marked as answer]
Why do use `held` to check if the item has a tag and then get the tag from `player.getItemInHand()` instead?

**Reply by Tofaa2** [reply]
> Why do use `held` to check if the item has a tag and then get the tag from `player.getItemInHand()` instead?

I'm not sure why that would make a difference since its just a memory point for the itemstack for both.

**Comment by Tofaa2** [comment]
Seems like this was in fact the issue. I'm not sure why but I'm marking it solved
